### ISSUE TITLE
Launch on primary display in Windows

### ISSUE THREAD

**Issue by mydoghasworms** [mamedev/mame#7922]
In Windows, mame does not respect the primary display setting to launch on the primary display.

I have a Windows 10 laptop with a monitor connected via HDMI. In the Windows settings, I have set the monitor to the primary display, but mame always launches on the laptop display.

I have tried the command line flags with little or no success, but it would be nice not to have to tinker with those and have mame just working "as expected".

There is a similar bug for Linux: #7532

**Comment by john-iv** [comment]
Run mame with the -v command to get the displays interrogated:


Mame dkong -v|find \\.\DISPLAY<file://./DISPLAY>

In my 3 monitor case it produces:
Video: Monitor 65539 = "\\.\DISPLAY1"
Video: Monitor 65537 = "\\.\DISPLAY2" (primary)
Video: Monitor 65541 = “\\.\DISPLAY3<file://./DISPLAY3>”

Then use mame dkong -screen \\.\DISPLAY2<file://./DISPLAY2> in this case.

I agree though, it would be nice for MAME to default to the primary.


From: Martin Ceronio ***@***.***>
Sent: Saturday, April 3, 2021 6:59 AM
To: mamedev/mame ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [mamedev/mame] Launch on primary display in Windows (#7922)


In Windows, mame does not respect the primary display setting to launch on the primary display.

I have a Windows 10 laptop with a monitor connected via HDMI. In the Windows settings, I have set the monitor to the primary display, but mame always launches on the laptop display.

I have tried the command line flags with little or no success, but it would be nice not to have to tinker with those and have mame just working "as expected".

There is a similar bug for Linux: #7532<https://github.com/mamedev/mame/issues/7532>

**Comment by DavidHaywood** [comment]
yeah MAME's behavior here is against the grain, especially as the default option is 'auto' which would suggest an intelligent choice.  In reality, it just always chooses 'screen1'

**Comment by Davidian1024** [comment]
For what it's worth I found a fix that's working for me on Ubuntu 20.04.6.  Maybe it'll work for you as well.
https://github.com/mamedev/mame/issues/7532#issuecomment-1713880191

**Comment by Davidian1024** [comment]
This might be fixed as of 0.259

**Comment by birdybro** [comment]
I can confirm it still is not fixed if you are in windowed mode. Windowed mode always launches on Display 1 in Windows no matter if it's primary at least for me.

The other problem is that it doesn't remember which display I dragged it to, which is how the vast majority of windowed applications in Windows behave. How it should behave is that when I drag it to Display 2 and close the application, if I open it again it should start on Display 2 next time.

Launching with special commandline parameters is not a real solution as the current UX is not idiomatic with how desktop apps commonly behave in windowed mode for the past 2 decades.

My current workaround for reference in windows is just make a shortcut to the application and make the target `C:\mame\mame.exe -screen \\.\DISPLAY2` but this is not ideal as now there is a console window stuck open since apparently any commandline parameters in windows keeps a console window open even from a shortcut. Not something I expected either coming from how other windows apps behave for the past... 2 decades or so. The default in windows ought to be that the console window is hidden when launching from a shortcut that has commandline params in it.

**Comment by rb6502** [comment]
As far as I'm aware none of the devs that handle this stuff have a multi-monitor setup to work on these issues.  Personally I switched from a dual-screen setup to a larger single 4K monitor years ago, and that or an ultrawide is pretty common now as replacements for multiple 1080P or lower displays.  If someone reading this is up on their Petzold and can figure out what's happening it would make some people very happy though.
